06-events-practice/2-tooltip/index.js

```diff
@@ -1,6 +1,65 @@
 class Tooltip {
-  initialize () {
+  static instance = null;
 
+  constructor() {
+    if (Tooltip.instance) {
+      return Tooltip.instance; 
+    }
+
+    Tooltip.instance = this;
+
+    this.element = null; 
+    this.targetElement = null; 
+  }
+
+  initialize() {
+    const container = document.createElement('div');
+    container.innerHTML = `<div id="container" data-tooltip="bar-bar-bar">
+      Aperiam consectetur dignissimos dolores ex mollitia.
+    </div>`;
+
+    this.targetElement = container.firstElementChild; 
+    this.element = document.createElement('div'); 
+
+    document.body.append(container); 
+    document.body.append(this.element); 
+
+    this.addEventListeners();
+  }
+
+  addEventListeners() {
+    this.targetElement.addEventListener('pointerover', this.showTooltip.bind(this));
+    this.targetElement.addEventListener('pointerout', this.hideTooltip.bind(this));
+  }
+
+
+  showTooltip() {
```

Comment:
handleDocumentPointerover(e) {
  if (!e.target.hasAttribute('data-tooltip')) {
   return;
   }
  const tooltipText = this.e.target.getAttribute('data-tooltip');
  this.element.textContent = tooltipText;
  this.document.body.appendChild(this.element)


}

Reply:
```
handleDocumentPointermove(e) {
    if (this.element.isConected) {
      this.element.style.left = e.pageX + 'px';
      this.element.style.top = e.pageY + 'px';
   }
}
```